spec/integration/data_stream/_dev/test/system/config.spec.yml

```
@@ -34,6 +34,15 @@ spec:
           enum:
             - docker
           default: docker
+        type:
```

Nit. Maybe we can be more explicit on the name of this parameter. But I don't have a better idea for this, maybe `base_image`, or `base_image_type`?
```suggestion
        base_image:
```

We have to keep in mind that this should be runtime-agnostic, but I think "image" is agnostic enough.

---

I was thinking in that and I had the same doubts.
I was wondering what would mean if at some point it is supported as runtime VMs.
Would it be used docker image inside that VM too ?

---

> I was wondering what would mean if at some point it is supported as runtime VMs.
> Would it be used docker image inside that VM too ?

I don't think so. Ideally we would install the agent directly in the VMs, or use an image that has the agent installed as a package or from tar/zip.
For the case of "complete" maybe we need to provide a different base image that includes the dependencies.
In that sense `base_image` would be fine and agnostic enough. It would not refer specifically to a docker image, but to a image in the general sense.

Another approach we could follow is to define this as dependencies or dependency sets:
```
depends:
  - synthetic_testing
```

This indicates the runtime that the test depends on the agent running on a system with capability to run synthetic tests. This is resolved in the docker runtime by using the complete image, but could be resolved in other systems in different ways.

---

> In that sense base_image would be fine and agnostic enough. It would not refer specifically to a docker image, but to a image in the general sense.

Using `base_image` looks good for me.

It would be like defining which kind of elastic-agent artifact/tar/zip it is required for that system test in a given package.

> Another approach we could follow is to define this as dependencies or dependency sets

This approach would allow developers to be more detailed to specify the requirements of each test. However, not sure if that could be too fine grained and therefore complicate the logic to select the required artifact/zip/docker.

---

> > Another approach we could follow is to define this as dependencies or dependency sets
> 
> This approach would allow developers to be more detailed to specify the requirements of each test. However, not sure if that could be too fine grained and therefore complicate the logic to select the required artifact/zip/docker.

Agree, so far we have only had two kinds of images and I don't think we are going to have many more cases.